Foundation/NSLock.swift

```diff
@@ -234,6 +234,8 @@ open class NSRecursiveLock: NSObject, NSLocking {
         super.init()
 #if os(Windows)
         InitializeCriticalSection(mutex)
+        InitializeConditionVariable(timeoutCond)
```

Comment:
Test?

Reply:
Oh. Just realized there is no test for locks. It would be helpful for sure. Should I add test to this particular PR?

Reply:
Can you add it in a subsequent PR? My suggestion: I would add the test with this reverted, make sure it fails and then add this back in.

Reply:
Sounds good, will do.